main/recording-history.js

```diff
@@ -0,0 +1,314 @@
+        '-movflags',
+        'faststart',
+        outputPath
+      ]);
+
+      return outputPath;
+    } catch {
+
+    }
+  }
+}];
+
+const handleCorruptRecording = async (recording, error) => {
+  const options = {
+    title: 'Kap didn\'t shut down correctly',
+    detail: `Looks like Kap crashed during a recording. We were able to locate the file. Unfortunately it appears to be corrupt.\n\n${error}`,
+    cancelId: 0,
+    defaultId: 2,
+    buttons: [
+      'Close',
+      {
+        label: 'Copy Error',
+        action: () => {
+          clipboard.writeText(error);
+        }
+      },
+      {
+        label: 'Show in Finder',
+        action: () => {
+          shell.showItemInFolder(recording.filePath);
+        }
+      }
+    ]
+  };
+
+  const applicableErrors = knownErrors.filter(({test}) => test(error));
+
+  if (applicableErrors.length > 0) {
+    options.message = 'We can attempt to repair the recording';
+    options.defaultId = 3;
+    options.buttons.push({
+      label: 'Attempt to Fix',
+      activeLabel: 'Attempting to Fix…',
+      action: async (_, updateUi) => {
+        for (const {fix} of applicableErrors) {
+          // eslint-disable-next-line no-await-in-loop
+          const outputPath = await fix(recording.filePath);
+
+          if (outputPath) {
+            addRecording({
+              filePath: outputPath,
+              name: recording.name,
+              date: new Date().toISOString()
+            });
+
+            return updateUi({
+              message: 'The recording was repaired successfully',
+              defaultId: 2,
+              buttons: [
+                'Close',
+                {
+                  label: 'Show in Finder',
+                  action: () => {
+                    shell.showItemInFolder(outputPath);
+                  }
+                },
+                {
+                  label: 'Show in Editor',
+                  action: () => openEditorWindow(outputPath, {recordingName: recording.name})
+                }
+              ]
+            });
+          }
+        }
+
+        return updateUi({
+          message: 'We were unable to repair the recording',
+          defaultId: 2,
+          buttons: [
+            'Close',
+            {
+              label: 'Copy Error',
+              action: () => {
+                clipboard.writeText(error);
+              }
+            },
+            {
+              label: 'Open in Finder',
+              action: () => {
+                shell.showItemInFolder(recording.filePath);
+              }
+            }
+          ]
+        });
+      }
+    });
+  } else {
+    const Sentry = require('./utils/sentry');
+    if (Sentry.isSentryEnabled) {
+      // Collect info about possible unknown errors, to see if we can implement fixes using ffmpeg
+      Sentry.captureException(new Error(`Corrupt recording: ${error}`));
+    }
+  }
+
+  return showDialog(options);
+};
+
+const checkForActiveRecording = async () => {
```

Comment:
Can you add a comment on what it returns? Because the prefix is `check` and not `has`, it's not clear what it would return.

Reply:
Returns true if there was a recording, and false otherwise. I'll rename to `has`. The return type is only used on start-up to control whether we show the cropper or not